boltdesignsystem/bolt
# Hotfix: Headline / Blockquote Text Alignment Regression Fix

## Jira
http://vjira2:8080/browse/BDS-2074

## Summary
Addresses a regression in last week's `v2.20.0` release in which the new Numbered Bullet feature added to Headline component inadvertently caused the component's text alignment to not work as expected in certain situations.

## Details
1. While Headline component's horizontal alignment (via the component's `align` prop) continues to work as expected, there were unfortunate side-effects from the globally applied `display: flex;` CSS update that caused text alignment to stop working in the Blockquote component + when the `u-bolt-text-align-X` utility class was used.

This PR works around that issue by only adding the `display: flex;` styles + related `c-bolt-headline__text` inner text wrapper in the specific use case that required both of those updates in the first place -- when a Headline is displaying the new Numbered Bullet option.

2. This PR also ensures that the `c-bolt-headline__text` wrapper continues to be conditionally added when the `quoted` version of Headline is used (which the original updates in v2.20.0 addressed)

3. This PR also fixes a separate issue related to this regression with Blockquote quote styles not rendering as expected once the page's JavaScript kicks in 

4. Finally, this PR cherry picks the one critical update from #1793 to ensure CI builds continue to work as expected.

## How to test
1. Confirm new text alignment-related VRT snapshots added for Blockquote look correct
2. Compare before and after links for Headline, Blockquote, and the referenced utility class instance to confirm these updates fix the visual regressions.

### Headline demos are visually similar [in v2.20.0](https://v2-20-0.boltdesignsystem.com/pattern-lab/?p=viewall-components-headline) compared to [this PR](https://hotfix-blockquote-alignment-fix-clean.boltdesignsystem.com/pattern-lab/?p=viewall-components-headline)
- [ ] Confirm that the Headline is visually as good or better as it was in the v2.20.0 release
- [ ] Confirm that the quotes that display when the Headline's `quoted` option is used show up as expected
- [ ] Confirm that the Right Icon in Headline is still inline when text wraps on smaller screens

### Blockquote text alignment [broken in v2.20.0](https://v2-20-0.boltdesignsystem.com/pattern-lab/?p=components-blockquote-alignItems-variation) is [now fixed](https://hotfix-blockquote-alignment-fix-clean.boltdesignsystem.com/pattern-lab/?p=components-blockquote-alignItems-variation)
- [x] Confirm that the Headline's text alignment usage within Blockquote is fixed in this PR when compared to the previous v2.20.0 release

### The broken Utility class alignment for "Add Mission" [in v2.20.0](https://v2-20-0.boltdesignsystem.com/pattern-lab/?p=blueprints-dashboard-add-mission) is [now fixed](https://hotfix-blockquote-alignment-fix-clean.boltdesignsystem.com/pattern-lab/?p=blueprints-dashboard-add-mission)
- [x] Confirm that the Headline component's text alignment is now fixed with this updates from this PR.
